desc/optimize/least_squares.py

```diff
@@ -208,14 +208,12 @@ def lsqtr(  # noqa: C901
     # conngould : norm of the cauchy point, as recommended in ch17 of Conn & Gould
     # scipy : norm of the scaled x, as used in scipy
     # mix : geometric mean of conngould and scipy
+    scipy = jnp.linalg.norm(x * scale_inv / v**0.5)
```

Comment:
I will be annoying but I don't like the idea of having `scipy` as variable name, especially since this file is mostly optimization stuff and future we might add some actual scipy functions.

Maybe something like `tr_scipy`?